### DISCUSSION TITLE
Try out Crossplane v1.14.0-rc.1

### DISCUSSION THREAD

**Discussion by turkenh** [crossplane/crossplane#4906] [General]
Crossplane version `v1.14.0-rc.1` is a release candidate intended to collect input from the community and offer users an opportunity to experiment with Crossplane in non-production environments before the official release of version `v1.14.0`. 

> [!WARNING]  
> This is a pre-release; do not use it in production environments!

To install Crossplane with this release:

```shell
helm repo add crossplane-stable https://charts.crossplane.io/stable --force-update
helm install crossplane --namespace crossplane-system --create-namespace crossplane-stable/crossplane --devel
```

To install Crossplane Cli with this release:

```shell
curl -sL https://raw.githubusercontent.com/crossplane/crossplane/v1.14.0-rc.1/install.sh | XP_VERSION=v1.14.0-rc.1 sh
```

The detailed release notes will be provided when the version `v1.14.0` is officially launched. In the meantime, you can review the commit changes made since version 1.13.0 with: https://github.com/crossplane/crossplane/compare/v1.13.0...v1.14.0-rc.1

For a high-level tour of the new features in Crossplane 1.14, see this gist from community member @stevendborrelli: [gist.github.com/stevendborrelli/f94d796076490010b4f03dede9dd886d](https://gist.github.com/stevendborrelli/f94d796076490010b4f03dede9dd886d)

<hr /><em>This discussion was created from the release <a href='https://github.com/crossplane/crossplane/releases/tag/v1.14.0-rc.1'>v1.14.0-rc.1</a>.</em>

**Comment by thoec** [comment]
Tried to install v1.14.0-rc.1. Got the following error:

```
$ helm install crossplane --namespace crossplane-system --create-namespace crossplane-stable/crossplane --devel
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(Deployment.spec.template.spec.containers[0].startupProbe.tcpSocket): unknown field "failureThreshold" in io.k8s.api.core.v1.TCPSocketAction, ValidationError(Deployment.spec.template.spec.containers[0].startupProbe.tcpSocket): unknown field "periodSeconds" in io.k8s.api.core.v1.TCPSocketAction]

```
(Stable release works fine to install: helm install crossplane --namespace crossplane-system --create-namespace crossplane-stable/crossplane)

$helm version
version.BuildInfo{Version:"v3.8.2", GitCommit:"6e3701edea09e5d55a8ca2aae03a68917630e91b", GitTreeState:"clean", GoVersion:"go1.17.5"}

On K8s 1.28

Looks like an indentation error in deployment.yaml?

```
startupProbe:
  tcpSocket:
    port: readyz
    failureThreshold: 30
    periodSeconds: 2
```
I guess failureThreshold and periodSeconds should be aligned with tcpSocket?

**Reply by phisco** [reply]
Can you share the full values.yaml?

**Reply by thoec** [reply]
I used the default values included with the chart. I fixed the issue locally (corrected the indentation) and managed to install the chart using my local copy.

**Reply by phisco** [reply]
But looking at the template I'd say you are right, strange our pipeline didn't catch it 🤔

**Reply by thoec** [reply]
Downloaded the chart by doing: helm pull crossplane-stable/crossplane --devel

Yes, agree :-)

**Reply by turkenh** [reply]
I cannot reproduce. Just tried with v1.28, but no luck. 
Any idea what is different here, helm version maybe, I have `v3.10.1` ?

**Reply by turkenh** [reply]
I think there is a change in behavior on how helm handles this, because with `v3.10.1` and `v3.13.1`, installation succeeds but found the following as warning on top of the logs:

```
W1027 15:25:09.222877   92835 warnings.go:70] unknown field "spec.template.spec.containers[0].startupProbe.tcpSocket.failureThreshold"
W1027 15:25:09.222891   92835 warnings.go:70] unknown field "spec.template.spec.containers[0].startupProbe.tcpSocket.periodSeconds"
```

Thanks @thoec for reporting, great catch 🙌

**Reply by turkenh** [reply]
Looks like this (not failing for invalid schema) is a helm bug with an open PR from last week: https://github.com/helm/helm/pull/12502
Once it is fixed (and we upgrade to latest helm in our e2e's) we could catch this kind of issues automatically. 

...hmm, it is just disabling/enabling the warning but not control failing fast 🤔

**Reply by turkenh** [reply]
PR for the fix: https://github.com/crossplane/crossplane/pull/4919
